compat/maven-settings-builder/src/main/java/org/apache/maven/settings/building/DefaultSettingsBuilder.java

```diff
@@ -257,27 +256,16 @@ public Object execute(String expression, Object value) {
     }
 
     private List<Server> serversByIds(List<Server> servers) {
-
-        if (servers.stream().allMatch(server -> server.getIds().isEmpty())) {
-            return servers;
-        }
-
         return servers.stream()
-                .flatMap(server -> {
-                    List<String> ids = server.getIds();
-                    if (ids.isEmpty()) {
-                        return Stream.of(server);
-                    } else {
-                        return Stream.concat(Stream.of(server), ids.stream().map(id -> newServer(server, id)));
-                    }
-                })
-                .collect(Collectors.toList());
+                .flatMap(server -> Stream.concat(
+                        Stream.of(server), server.getIds().stream().map(id -> serverAlias(server, id))))
+                .toList();
     }
 
-    private Server newServer(Server server, String id) {
-        return new Server(org.apache.maven.api.settings.Server.newBuilder(server.getDelegate(), false)
+    private Server serverAlias(Server server, String id) {
+        return new Server(org.apache.maven.api.settings.Server.newBuilder(server.getDelegate(), true)
                 .id(id)
-                .ids(Collections.emptyList())
```

Comment:
assuming this is redundant, as we work on new instance; things must be init correctly by default.
This is not our concern.
Here we only call, to override existing value. This seems to apply default on default.

Test should require or deletion.

Reply:
new instance is based on given server with ids